src/FormField/Upload.php

```diff
@@ -87,6 +87,26 @@ public function init()
         }
     }
 
+    /**
+     * Allow to set file name and file id.
+     *  - Value will be display to user, ex: file name.
+     *  - fileId is the file id send when using onDelete callback.
+     *      If null, onDelete receive the file name.
+     *
+     * @param mixed|null  $value  // Field value, display to user.
+     * @param string|null $fileId // Field id.
+     *
+     * @return $this|void
+     */
+    public function set($value, $fileId = null)
+    {
+        if ($fileId) {
+            $this->setFileId($fileId);
```

Comment:
well i think there is still inconsistency throughout here. I'll give analogy with form fields:

$form->addField('name');

Now - form appears as "Name" if caption is not specified. However name is mandatory and must be always set. 


In your case, it would be understandable if the 'fileId' would be the primary identifier and the only required parameter, not only here, everywhere. "original_filename" is an optional parameter which can be passed from the onUpload() but it would be insignificant and wouldn't be stored anywhere.  Maybe in the future we also add "icon" or "size" parameters which are also insignificant and are not really needed at form submission.


In the `demo` you can use this callback:

```
move_uploaded_file([..], 'files/'.$original_filename);
return $original_filename;
```

so in this case $original_filename acts as ID and is passed to the form which can, for example, do some more stuff with the file.


Probably a refactoring is needed, which we can do in the future, but it's important that we swap the arguments to set() here.